airflow/ui/src/components/FlexibleForm/FieldDateTime.tsx

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Input } from "@chakra-ui/react";
+
+import type { FlexibleFormElementProps } from ".";
+
+export type DateTimeElementProps = {
+  readonly subType: string;
+} & FlexibleFormElementProps;
+
+const typeToPlaceholder: Record<string, string> = {
+  date: "yyyy-mm-dd",
+  "datetime-local": "yyyy-mm-ddThh:mm",
+  time: "hh:mm",
+};
+
+export const FieldDateTime = ({ name, param, subType }: DateTimeElementProps) => (
```

Comment:
```suggestion
export const FieldDateTime = ({ name, param, ...rest }: FlexibleFormElementProps & InputProps) => (
```

we can just use the existing InputProps. And the pre-existing placeholders are actually fine:

<img width="843" alt="Screenshot 2025-01-14 at 10 03 21 PM" src="https://github.com/user-attachments/assets/946f46df-a415-4496-95d7-5cab138705c7" />

Reply:
These are US defaults for me, are the placeholders different for you based on your locale?

Reply:
Nope, the placeholders are hard-coded. Do you know any chakra localizable option?

I just checked but also using @orange_digital/chakra-datepicker is not a better option in my view... short research tells me... or I am blind - don't see better components available?
Maybe we start the the US placeholders and rework later? Otherwise ideas welcome. (not an expert in this :-( are there localization libs?)

Reply:
Just noticed that the placehodlers not needed at all for date+time pickers. They are generated by the browser.

I removed the complexity.

And also realized that if I switch my system locale to German (I use EN-US per default) then in Firefox the date and time placeholders are correct.